Dockerfile

```diff
@@ -29,6 +42,8 @@ ADD . /app
 EXPOSE 8080
 ENV DJANGO_SETTINGS_MODULE=fee_calculator.settings.docker
 
-# need secret_key env var here
-RUN python3 manage.py collectstatic
-CMD python3 manage.py migrate --no-input && uwsgi --ini uwsgi.ini
+RUN python3 manage.py migrate --no-input \
+    && python3 manage.py loadalldata \
+    && python3 manage.py collectstatic --no-input
```

Comment:
If you did all this stuff as part of the `CMD`, would you then not need to supply a secret key as part of the build process?

Reply:
How worth it is it? I mean these commands would add I assume approx 30 seconds to start up time - how long does it take to spin up a new pod otherwise?

Reply:
It's entirely up to you, but you might want to consider not running migrations and `collectstatic` as part of container startup, as that means every replica will run those commands on every scale event, restart etc, which is likely inefficient and unnecessary.

An alternative approach is to use a separate [`Job` resource](https://kubernetes.io/docs/concepts/workloads/controllers/jobs-run-to-completion/) to handle tasks like that - an example of that approach can be seen in the reference app's [`Dockerfile`](https://github.com/ministryofjustice/cloud-platform-reference-app/blob/master/Dockerfile#L14) and [`job.yaml`](https://github.com/ministryofjustice/cloud-platform-reference-app/blob/master/helm_deploy/django-app/templates/job-migration.yaml).

Reply:
@kerin we briefly explored using a `Job` but it complicated the `Pod` lifecycle so we decided to go with baking them in the image.